lib/starlark_compiler/ast.rb

```diff
@@ -101,13 +101,38 @@ def initialize(str)
       end
     end
 
+    class VariableReference < Node
+      attr_reader :var
+      def initialize(var)
+        unless var.is_a?(::String)
+          raise ''"
+Only string type is allowed as a variable reference, got #{var.class}
+"''
+        end
+
+        @var = var
+      end
+    end
+
     class Array < Node
       attr_reader :elements
       def initialize(elements)
         @elements = elements.map(&method(:node))
       end
     end
 
+    class VariableAssignment < Node
+      attr_reader :name, :var
+      def initialize(name, var)
+        @name = name
+        if [VariableAssignment].include?(var.class)
```

Comment:
Potentially worth marking nodes as statements vs expressions, and only allow expressions on the RHS?

Reply:
I think that makes more sense long term

I don't write much ruby, so please let me know if theres a better way to implement this than i did (there probably is)